File: setup_dirs.js
const fs = require('fs');
const path = require('path');

const basePath = 'c:\\Users\\mukerana\\Documents\\VS_Code\\github_repo\\crt_repo';

// Create directories
const agentsDir = path.join(basePath, '.github', 'agents');
const promptsDir = path.join(basePath, '.github', 'prompts');

fs.mkdirSync(agentsDir, { recursive: true });
fs.mkdirSync(promptsDir, { recursive: true });

console.log('✓ Created directories:');
console.log('  - .github/agents');
console.log('  - .github/prompts');

// Read source files and copy to destination
const file1Source = path.join(basePath, 'e2e-uat-pipeline.agent.md');
const file1Dest = path.join(agentsDir, 'e2e-uat-pipeline.agent.md');
const file2Source = path.join(basePath, 'create-e2e-uat-pipeline.prompt.md');
const file2Dest = path.join(promptsDir, 'create-e2e-uat-pipeline.prompt.md');

fs.copyFileSync(file1Source, file1Dest);
fs.copyFileSync(file2Source, file2Dest);

console.log('\n✓ Copied files:');
console.log(`  - ${file1Dest}`);
console.log(`  - ${file2Dest}`);

// Verify files
console.log('\n✓ Verification:');
if (fs.existsSync(file1Dest)) {
  const stats1 = fs.statSync(file1Dest);
  console.log(`  File 1: FOUND (${stats1.size} bytes)`);
}
if (fs.existsSync(file2Dest)) {
  const stats2 = fs.statSync(file2Dest);
  console.log(`  File 2: FOUND (${stats2.size} bytes)`);
}

// Show directory structure
console.log('\n✓ Directory Structure:');
const showTree = (dir, prefix = '') => {
  const files = fs.readdirSync(dir);
  files.forEach((file, index) => {
    const fullPath = path.join(dir, file);
    const isLast = index === files.length - 1;
    const connector = isLast ? '└── ' : '├── ';
    console.log(prefix + connector + file);
    
    if (fs.statSync(fullPath).isDirectory() && file !== 'workflows') {
      const nextPrefix = prefix + (isLast ? '    ' : '│   ');
      showTree(fullPath, nextPrefix);
    }
  });
};

showTree(path.join(basePath, '.github'));

File: setup_dirs.py
import os
import shutil

base_path = r'c:\Users\mukerana\Documents\VS_Code\github_repo\crt_repo'

# Create directories
agents_dir = os.path.join(base_path, '.github', 'agents')
prompts_dir = os.path.join(base_path, '.github', 'prompts')

os.makedirs(agents_dir, exist_ok=True)
os.makedirs(prompts_dir, exist_ok=True)

print('✓ Created directories:')
print('  - .github/agents')
print('  - .github/prompts')

# Copy files
file1_source = os.path.join(base_path, 'e2e-uat-pipeline.agent.md')
file1_dest = os.path.join(agents_dir, 'e2e-uat-pipeline.agent.md')
file2_source = os.path.join(base_path, 'create-e2e-uat-pipeline.prompt.md')
file2_dest = os.path.join(prompts_dir, 'create-e2e-uat-pipeline.prompt.md')

shutil.copy2(file1_source, file1_dest)
shutil.copy2(file2_source, file2_dest)

print('\n✓ Copied files:')
print(f'  - {file1_dest}')
print(f'  - {file2_dest}')

# Verify files
print('\n✓ Verification:')
if os.path.exists(file1_dest):
    size1 = os.path.getsize(file1_dest)
    print(f'  File 1 (.github/agents/e2e-uat-pipeline.agent.md): FOUND ({size1} bytes)')
if os.path.exists(file2_dest):
    size2 = os.path.getsize(file2_dest)
    print(f'  File 2 (.github/prompts/create-e2e-uat-pipeline.prompt.md): FOUND ({size2} bytes)')

# Show directory structure
print('\n✓ Directory Structure:')
def show_tree(dir_path, prefix=''):
    try:
        items = sorted(os.listdir(dir_path))
        for i, item in enumerate(items):
            if item == '__pycache__':
                continue
            full_path = os.path.join(dir_path, item)
            is_last = i == len(items) - 1
            connector = '└── ' if is_last else '├── '
            print(prefix + connector + item)
            
            if os.path.isdir(full_path) and item not in ['workflows', '.git']:
                next_prefix = prefix + ('    ' if is_last else '│   ')
                show_tree(full_path, next_prefix)
    except Exception as e:
        print(f'Error reading directory: {e}')

github_path = os.path.join(base_path, '.github')
show_tree(github_path)
